cmd/regctl/digest.go

```diff
@@ -1,20 +1,36 @@
 package main
 
 import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+
 	"fmt"
 	"io"
-	"os"
 
 	// crypto libraries included for go-digest
 	_ "crypto/sha256"
 	_ "crypto/sha512"
 
+	"github.com/klauspost/compress/zstd"
 	"github.com/opencontainers/go-digest"
 	"github.com/spf13/cobra"
 )
 
+var gzipMagicHeader = []byte{'\x1f', '\x8b'}
+var zstdMagicHeader = []byte{'\x28', '\xb5', '\x2f', '\xfd'}
+
+type compression string
+
+const (
+	compressionGzip compression = "gzip"
+	compressionZstd compression = "zstd"
+	compressionNone compression = "none"
+)
+
 type digestCmd struct {
-	rootOpts *rootCmd
+	rootOpts   *rootCmd
+	decompress string
```

Comment:
Lets make this a `bool` and use automatic detection when true.

Reply:
Done